axelrod/result_set.py

```diff
@@ -28,314 +33,297 @@ def wrapper(*args):
-        else:
-            self.players, self.repetitions = players, repetitions
-
-        self.nplayers = len(self.players)
-
-        self._build_empty_metrics()
-        self._build_score_related_metrics(progress_bar=progress_bar)
-
-    def create_progress_bar(self, desc=None):
-        """
-        Create a progress bar for a read through of the data file.
-
-        Parameters
-        ----------
-            desc : string
-                A description.
+                The number of repetitions of each match. If not know will be
+                efficiently read from file.
+            processes : integer
+                The number of processes to be used for parallel processing
         """
-        return tqdm.tqdm(total=self.num_matches, desc=desc)
+        self.filename = filename
+        self.players, self.repetitions = players, repetitions
+        self.num_players = len(self.players)
 
-    def _update_players(self, index_pair, players):
-        """
-        During a read of the data, update the internal players dictionary
+        if progress_bar:
+            self.progress_bar = tqdm.tqdm(total=25,
+                                          desc="Analysing")
 
-        Parameters
-        ----------
+        df = dd.read_csv(filename)
+        dask_tasks = self._build_tasks(df)
 
-            index_pair : tuple
-                A tuple of player indices
-            players : tuple
-                A tuple of player names
-        """
-        for index, player in zip(index_pair, players):
-            if index not in self.players_d:
-                self.players_d[index] = player
+        if processes == 0:
+            processes = cpu_count()
 
-    def _update_repetitions(self, index_pair, nbr=1):
-        """
-        During a read of the data, update the internal repetitions dictionary
+        out = self._compute_tasks(tasks=dask_tasks, processes=processes)
 
-        Parameters
-        ----------
+        self._reshape_out(out)
 
-            index_pair : tuple
-                A tuple of player indices
-            nbr : integer
-                The number of repetitions
-        """
-        try:
-            self.repetitions_d[index_pair] += nbr
-        except KeyError:
-            self.repetitions_d[index_pair] = nbr
+        if progress_bar:
+            self.progress_bar.close()
 
-    def _build_repetitions(self):
+    def _reshape_out(self, out):
         """
-        Count the number of repetitions
-
-        Returns
-        -------
-
-            repetitions : int
-                The number of repetitions
+        Reshape the various pandas series objects to be of the required form and
+        set the corresponding attributes.
         """
-        repetitions = max(self.repetitions_d.values())
 
-        del self.repetitions_d  # Manual garbage collection
-        return repetitions
+        (mean_per_reps_player_opponent_df,
```

Comment:
Is there a reason not to pass these in explicitly as parameters? Or to use `*args` instead?

Reply:
Yeah that's way better :+1: